tests/contrib/hahelpers/test_apache_utils.py

```diff
@@ -131,3 +131,129 @@ def test_retrieve_ca_cert_no_file(self, _isfile):
                 apache_utils.retrieve_ca_cert('mycertfile'),
                 None)
             self.assertFalse(_open.called)
+
+    def test_validate_cert_valid(self):
+        cert = '''-----BEGIN CERTIFICATE-----
+MIICKTCCAdCgAwIBAgIRAJ94pSPwW8j47TxyEbuy11cwCgYIKoZIzj0EAwIwODES
+MBAGA1UEChMJTG9jYWxob3N0MSIwIAYDVQQDExlMb2NhbGhvc3QgSW50ZXJtZWRp
+YXRlIENBMB4XDTIyMDYxNTIzMzgwMVoXDTIyMDYxNjIzMzkwMVowFDESMBAGA1UE
+AxMJbG9jYWxob3N0MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAExkWwyfOpgiAh
+oClU2ETkZ+C2Q5FD81f4nk6IDRCZ0JEhXl6b5t1z02F0+aA3FZRkmlJBuGv9Oi+v
+AJ8R2Yh1BaOB3jCB2zAOBgNVHQ8BAf8EBAMCB4AwHQYDVR0lBBYwFAYIKwYBBQUH
+AwEGCCsGAQUFBwMCMB0GA1UdDgQWBBQXIcm+FtQbyWuRivPP9xTVgCIp6zAfBgNV
+HSMEGDAWgBSwrFttsNHlNYEjUKqvinnPVR/+UDAUBgNVHREEDTALgglsb2NhbGhv
+c3QwVAYMKwYBBAGCpGTGKEABBEQwQgIBAQQQdGVzdEBleGFtcGxlLmNvbQQrZFRH
+VC14NUtycHpRN1M3VnIxSVpnMVh4cVRuRi1INml3M1lqMVZoS1JxWTAKBggqhkjO
+PQQDAgNHADBEAiAyQVuIbUuYKcps8EolwLL7HVbwVMdK06rvOqEIcNnpCAIgdEFi
+8XsMn9hJmDiunM8RAd0aDVa1cC+aizu5mOgrBbc=
+-----END CERTIFICATE-----
+-----BEGIN CERTIFICATE-----
+MIIBzTCCAXKgAwIBAgIQcT2oOYh7TNlbbto/RdvZGTAKBggqhkjOPQQDAjAwMRIw
+EAYDVQQKEwlMb2NhbGhvc3QxGjAYBgNVBAMTEUxvY2FsaG9zdCBSb290IENBMB4X
+DTIyMDYxNTIzMzg0N1oXDTMyMDYxMjIzMzg0N1owODESMBAGA1UEChMJTG9jYWxo
+b3N0MSIwIAYDVQQDExlMb2NhbGhvc3QgSW50ZXJtZWRpYXRlIENBMFkwEwYHKoZI
+zj0CAQYIKoZIzj0DAQcDQgAEKpD0lYmyDLX+guoi2WQupkRpYydpkT0fihJ9Asqp
+bQ0j8SK6XOgG17wIIynR/PchZPBnTew8poi1TOV6Sf195KNmMGQwDgYDVR0PAQH/
+BAQDAgEGMBIGA1UdEwEB/wQIMAYBAf8CAQAwHQYDVR0OBBYEFLCsW22w0eU1gSNQ
+qq+Kec9VH/5QMB8GA1UdIwQYMBaAFL7pE7+4bPZwOYJeDceK/B2Ev4iSMAoGCCqG
+SM49BAMCA0kAMEYCIQC3NxTe3PxTiF0W1L39xrHvPvFCIHkI65ThQWM+upgNIAIh
+AJD2IG2GS7MCxD5EiG4Y3cMcPjye4qud4OduSbnA8UO4
+-----END CERTIFICATE-----
+'''.encode()
```

Comment:
It might be better to add a header to PEM as follows to be easier to review or revisit later on.

With `openssl x509 -subject -issuer`, this block can be:

```
subject=CN = localhost
issuer=O = Localhost, CN = Localhost Intermediate CA
-----BEGIN CERTIFICATE-----
MIICKTCCAdCgAwIBAgIRAJ94pSPwW8j47TxyEbuy11cwCgYIKoZIzj0EAwIwODES
MBAGA1UEChMJTG9jYWxob3N0MSIwIAYDVQQDExlMb2NhbGhvc3QgSW50ZXJtZWRp
YXRlIENBMB4XDTIyMDYxNTIzMzgwMVoXDTIyMDYxNjIzMzkwMVowFDESMBAGA1UE
AxMJbG9jYWxob3N0MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAExkWwyfOpgiAh
oClU2ETkZ+C2Q5FD81f4nk6IDRCZ0JEhXl6b5t1z02F0+aA3FZRkmlJBuGv9Oi+v
AJ8R2Yh1BaOB3jCB2zAOBgNVHQ8BAf8EBAMCB4AwHQYDVR0lBBYwFAYIKwYBBQUH
AwEGCCsGAQUFBwMCMB0GA1UdDgQWBBQXIcm+FtQbyWuRivPP9xTVgCIp6zAfBgNV
HSMEGDAWgBSwrFttsNHlNYEjUKqvinnPVR/+UDAUBgNVHREEDTALgglsb2NhbGhv
c3QwVAYMKwYBBAGCpGTGKEABBEQwQgIBAQQQdGVzdEBleGFtcGxlLmNvbQQrZFRH
VC14NUtycHpRN1M3VnIxSVpnMVh4cVRuRi1INml3M1lqMVZoS1JxWTAKBggqhkjO
PQQDAgNHADBEAiAyQVuIbUuYKcps8EolwLL7HVbwVMdK06rvOqEIcNnpCAIgdEFi
8XsMn9hJmDiunM8RAd0aDVa1cC+aizu5mOgrBbc=
-----END CERTIFICATE-----
subject=O = Localhost, CN = Localhost Intermediate CA
issuer=O = Localhost, CN = Localhost Root CA
-----BEGIN CERTIFICATE-----
MIIBzTCCAXKgAwIBAgIQcT2oOYh7TNlbbto/RdvZGTAKBggqhkjOPQQDAjAwMRIw
EAYDVQQKEwlMb2NhbGhvc3QxGjAYBgNVBAMTEUxvY2FsaG9zdCBSb290IENBMB4X
DTIyMDYxNTIzMzg0N1oXDTMyMDYxMjIzMzg0N1owODESMBAGA1UEChMJTG9jYWxo
b3N0MSIwIAYDVQQDExlMb2NhbGhvc3QgSW50ZXJtZWRpYXRlIENBMFkwEwYHKoZI
zj0CAQYIKoZIzj0DAQcDQgAEKpD0lYmyDLX+guoi2WQupkRpYydpkT0fihJ9Asqp
bQ0j8SK6XOgG17wIIynR/PchZPBnTew8poi1TOV6Sf195KNmMGQwDgYDVR0PAQH/
BAQDAgEGMBIGA1UdEwEB/wQIMAYBAf8CAQAwHQYDVR0OBBYEFLCsW22w0eU1gSNQ
qq+Kec9VH/5QMB8GA1UdIwQYMBaAFL7pE7+4bPZwOYJeDceK/B2Ev4iSMAoGCCqG
SM49BAMCA0kAMEYCIQC3NxTe3PxTiF0W1L39xrHvPvFCIHkI65ThQWM+upgNIAIh
AJD2IG2GS7MCxD5EiG4Y3cMcPjye4qud4OduSbnA8UO4
-----END CERTIFICATE-----
```